examples/kafka/kafka-ephemeral-oauth-single-keycloak-authz.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    version: 2.7.0
+    replicas: 1
+    listeners:
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+        authentication:
+          type: oauth
+          validIssuerUri: https://${SSO_HOST}/auth/realms/kafka-authz
+          jwksEndpointUri: https://${SSO_HOST}/auth/realms/kafka-authz/protocol/openid-connect/certs
+          userNameClaim: preferred_username
+          maxSecondsWithoutReauthentication: 3600
+          tlsTrustedCertificates:
+            - secretName: oauth-server-cert
+              certificate: sso.crt
+    authorization:
+      type: keycloak
+      clientId: kafka
+      tokenEndpointUri: https://${SSO_HOST}/auth/realms/kafka-authz/protocol/openid-connect/token
```

Comment:
We should not use variables. It should have some URL.

Reply:
For this one there really is no other way that I can see. What do you mean with 'some URL'?

Reply:
People go to the examples and use them. When they see an URL they will understand that they might need to change it.  But this is confusing as lot of them would expect this to be magically autofilled since it is some sort variable. The things in the examples folder cannot rely on the documentation to know how to be used. I think you can deal with it this way: Move the YAML into the docs and have it live there (but that means the Keycloak realm JSON makes no sense in examples either). Or you move the explanation on how to use the example from the docs into a README.md and place all of it somewhere in `examples/security/keycloak-authorization`.

Reply:
The problem is that in the example we point the user to Keycloak documentation for instaling Keycloak, and so we can't know in advance where it is installed. If we include some fake guaranteed-not-to-work hostname the install is guarranteed not to work and it will be even less obvious why it doesn't work.
What if I move `examples/kafka/kafka-ephemeral-oauth-single-keycloak-authz.yaml` to `examples/security/keycloak-authorization/kafka-ephemeral-oauth-single-keycloak-authz.yaml`, and `examples/security/keycloak/kafka-authz-realm.json` to `examples/security/keycloak-authorization/kafka-authz-realm.json` and add a README.md in that dir with some instructions, but at the same time leave the variable in the file?

Reply:
Well, as I said ... then it probably either should not be in the examples but only covered in the docs ... or the docs should move to some README.md in the examples. People see the examples without reading the docs so they need to make sense and be usable that way.

Reply:
To properly deploy is quite some amount of texts, but the full example is too big to be part of examples. So, either like you say these files become part of the documentation, which AFAICT has not been done before, so we need to include non-readable files (json, yaml) to be part of the documentaion? Or, I updated the PR with a halfway solution - added a README.md where a give some explanations but also link to the example in the documentation so it's clear to the user, that the instructions are more involved and that the specific files are part of the example in the documentation.
Would that be acceptable?

Reply:
How do you make sure it links to the right version of the documentation? You will probably need to add something to the Makefiles or something to link to master by default but update the link during releases. But apart from that it sounds reasonable.

Reply:
I moved examples files from `examples` to `packaging/examples` and added some script to `Makefile` and `packaging/examples/Makefile` so that during `make release` the links in examples README point to the correspondingly tagged strimzi.io documentation, and links in keycloak example in Using Strimzi doc point to the proper corresponding version tag of strimzi-kafka-oauth.

For an ordinary build the links in examples README point to `latest` docs and the links in keycloak example Using Strimzi doc point to the `main` branch of strimzi-kafka-oauth.

Reply:
I will review the code ... not sure this is correct. I will let you know if needed.